backend-lambda/index.js

```diff
@@ -25,33 +31,33 @@ export const handler = async (event, context) => {
   }
   let response;
 
-  if (event.path === "/products") {
+  if (path === "/products") {
     // get all products
-    if (event.httpMethod === "GET") {
+    if (httpMethod === "GET") {
       const limit = event.queryStringParameters.limit;
       const offset = event.queryStringParameters.offset;
       response = await productService.getAllProducts(pool, limit, offset);
     }
     // create a product
-    else if (event.httpMethod === "POST") {
-      const product = body;
-      response = await productService.createProduct(pool, product);
+    else if (httpMethod === "POST") {
+      response = await productService.createProduct(pool, body);
     }
   }
+
   // update a product
-  else if (event.path === "/product" && event.httpMethod === "PUT") {
-    response = await productService.updateProduct(pool, body);
-  }
-  // update a product as sold
-  else if (event.path === "/products/sold" && event.httpMethod === "PUT") {
+  else if (path.includes("/product/") && httpMethod === "PUT") {
+    await validate.validateProductId(event.pathParameters.product_id);
     response = await productService.updateProduct(pool, body.product_id, body);
   }
+
   // make items on floor
-  else if (event.path === "/products/to-floor" && event.httpMethod === "POST") {
+  else if (path === "/products/to-floor" && httpMethod === "POST") {
```

Comment:
You can delete this and everything having to do with to-floor

Reply:
Removed